frame/evm/precompile/withdraw/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::Currency;
+use sp_core::U256;
+use sp_runtime::{traits::UniqueSaturatedInto, DispatchError};
+use sp_std::marker::PhantomData;
+use sp_std::prelude::*;
+use sp_std::vec::Vec;
+
+use codec::Decode;
+use darwinia_evm::Trait;
+use darwinia_evm_primitives::Precompile;
+use evm::{Context, ExitError, ExitSucceed};
+
+type AccountId<T> = <T as frame_system::Trait>::AccountId;
+
+// WithDraw Precompile Contract, used to withdraw balance from evm account to darwinia account
+// address: 0000000000000000000000000000000000000005
+pub struct WithDraw<T: Trait> {
+	_maker: PhantomData<T>,
+}
+
+impl<T: Trait> Precompile for WithDraw<T> {
+	fn execute(
+		input: &[u8],
+		_target_gas: Option<usize>,
+		context: &Context,
+	) -> core::result::Result<(ExitSucceed, Vec<u8>, usize), ExitError> {
+		// Decode input data
+		let input = InputData::<T>::decode(&input).unwrap();
+
+		let helper = U256::from(10)
+			.checked_pow(U256::from(9))
+			.unwrap_or(U256::MAX);
+		let (context_value, _) = context.apparent_value.div_mod(helper);
+		let context_value = context_value.low_u128().unique_saturated_into();
+
+		// Here is a tricky area that needs to be clarified.
```

Comment:
So the address of `Withdraw` contract could be ignored when to calculate total supply  as the same as `0x0`.

Reply:
Yes, 0x5 will be a black hole account, similar to 0x0. This account needs to be skipped when calculating the total number of rings, otherwise it will result in a higher calculation than the actual value.